store/orm/orm.go

```diff
@@ -755,3 +756,39 @@ func (orm *ORM) BulkDeleteRuns(bulkQuery *models.BulkDeleteRunRequest) error {
 	}
 	return tx.Commit().Error
 }
+
+// Keys returns all keys stored in the orm.
+func (orm *ORM) Keys() ([]*models.Key, error) {
+	var keys []*models.Key
+	return keys, orm.DB.Find(&keys).Error
+}
+
+// FirstOrCreateKey returns the first key found or creates a new one in the orm.
+func (orm *ORM) FirstOrCreateKey(k *models.Key) error {
+	return orm.DB.FirstOrCreate(k).Error
+}
+
+// ClobberDiskKeyStoreWithDBKeys writes all keys stored in the orm to
+// the keys folder on disk, deleting anything there prior.
+func (orm *ORM) ClobberDiskKeyStoreWithDBKeys(keysDir string) error {
+	if err := os.RemoveAll(keysDir); err != nil {
```

Comment:
Do we need to remove keys we don't know about? Is it possible to just ignore them? Seems like it has the potential to be very destructive. Possibly worth erroring out instead. I know we talked about this before, don't think I realized the implication that we'd be completely deleting private keys.

Reply:
It makes things considerably simpler when syncing to do this. I have since changed the keys directory to `tempkeys` to communicate the ephemeral nature of the files in this folder. Let's talk about it more offline.